core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/decorator/NameserverPillarDecorator.java

```diff
@@ -56,19 +58,38 @@ private void decorateServicePillarWithDatalakeNameservers(StackDto stackDto, Map
             List<InstanceMetadataView> gatewayInstanceMetadata = getGatewayInstanceMetadata(datalakeStackOptional.get());
             String datalakeDomain = gatewayInstanceMetadata.get(0).getDomain();
             List<String> ipList = getDatalakeGatewayPrivateIps(gatewayInstanceMetadata);
+            validateIpList(ipList);
             servicePillar.put("forwarder-zones", new SaltPillarProperties("/unbound/forwarders.sls",
                     singletonMap("forwarder-zones", singletonMap(datalakeDomain, singletonMap("nameservers", ipList)))));
         } else {
             LOGGER.debug("Skip to add nameserver config for pillar because Datalake stack is not present.");
         }
     }
 
+    private static List<String> getKerberosNameServerIps(KerberosConfig kerberosConfig) {
+        return Lists.newArrayList(kerberosConfig.getNameServers().split(","))
+                .stream()
+                .filter(filterValidIps())
+                .collect(Collectors.toList());
+    }
+
     private List<String> getDatalakeGatewayPrivateIps(List<InstanceMetadataView> gatewayInstanceMetadata) {
         return gatewayInstanceMetadata.stream()
                 .map(InstanceMetadataView::getPrivateIp)
-                .filter(StringUtils::hasText)
-                .collect(Collectors.collectingAndThen(Collectors.toList(), Optional::of))
-                .filter(ipList -> !ipList.isEmpty())
-                .orElseThrow(() -> new CloudbreakServiceException("Unable to setup nameservers because there is no IP address present."));
+                .filter(filterValidIps())
+                .collect(Collectors.toList());
+    }
+
+    private static Predicate<String> filterValidIps() {
+        return ip -> StringUtils.hasText(ip) && !"null".equals(ip);
```

Comment:
you could use an IPv4 matcher which would be more safe then just checking for `null`